docs/guides/concurrency.md

```diff
@@ -219,10 +219,13 @@ This prints:
 ```
 Before receive
 Before send
+After send
 After receive
 ```
 
-First, the program spawns a fiber but doesn't execute it yet. When we invoke `channel.receive`, the main fiber blocks and execution continues with the spawned fiber. Then `channel.send(nil)` is invoked, and so execution continues at `channel.receive`, which was waiting for a value. Then the main fiber continues executing and finishes, so the program exits without giving the other fiber a chance to print "After send".
+First, the program spawns a fiber but doesn't execute it yet. When we invoke `channel.receive`, the main fiber blocks and execution continues with the spawned fiber. Then `channel.send(nil)` is invoked, Note that this `send` does not occupy space in the channel, This is because there is a `receive` invoked prior to the first `send`, so this `send` not got blocked. Fiber only switch out when blocked or executing to completion, so, will continue the codes after `send`, switch back to main fiber when finish `puts "After send"`.
```

Comment:
```suggestion
First, the program spawns a fiber but doesn't execute it yet. When we invoke `channel.receive`, the main fiber blocks and execution continues with the spawned fiber. Then `channel.send(nil)` is invoked. Note that this `send` does not occupy space in the channel because there is a `receive` invoked prior to the first `send`, `send` is not blocked. Fibers only switch out when blocked or executing to completion. So the spawned fiber will continue after `send`, and execution will switch back to main fiber once `puts "After send"` is executed.
```